openid_connect.js

```diff
@@ -7,7 +7,43 @@ var newSession = false; // Used by oidcAuth() and validateIdToken()
 
 export default {auth, codeExchange, validateIdToken, logout};
 
+function sleep(ms) {
+    return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+// If the ID token has not been synced yet, poll the variable every 100ms until
+// get a value or after a timeout.
+async function waitForSessionSync(r, timeout, checkInterval) {
```

Comment:
Version without `async`:
```javascript
function waitForSessionSync(r, timeout, checkInterval) {
    return new Promise(function (resolve, reject) {
        let timePass = 0;
        function check() {
	    try {
                if (r.variables.session_jwt) {
                    resolve(true);
                } else if (timePass >= timeout) {
                    resolve(false);
                } else {
                    timePass += checkInterval;
                    setTimeout(check, checkInterval);
                }
            } catch(e) {
	        reject(e);
            }
	}
        check();
    });
}
```

Reply:
Thank you @alebastr, Replaced with the option suggested by Pavel. Indeed, the use of async functions is not justified in our case.